### PR TITLE
Finished documenting PoketchData

### PR DESCRIPTION
**Pull request by TrainerRiley**

### CODE REVIEW

File: include/poketch_data.h [hunk 1]
```diff
@@ -50,19 +50,21 @@ enum PoketchScreenColor {
 };
 
 enum {
```
Comment (cbt6) [inline include/poketch_data.h:52]: Why are these values in an enum? They don't seem related to each other.

File: include/poketch_data.h [hunk 2]
```diff
@@ -273,13 +291,70 @@ void PoketchData_SetMapMarker(PoketchData *poketchData, int index, u8 x, u8 y);
...
+void PoketchData_DotArtData(const PoketchData *poketchData, u8 *dotArtData);
+
+/**
+ * Updates the PoketchData's stored dotArt data and marks the Modified flag as TRUE. 
+ * The function will copy POKETCH_DOTART_SIZE_BYTES bytes of data at the given pointer.
+ *
+ * @param poketchData:      The Poketch data to update.
+ * @param dotArtData:       The dot art data to copy.
+ */
+void PoketchData_SetDotArtData(PoketchData *poketchData, const u8 *dotArtData);
```
Comment (cbt6) [inline include/poketch_data.h:320]:
```suggestion
void PoketchData_ModifyDotArtData(PoketchData *poketchData, const u8 *src);
```

File: include/poketch_data.h [hunk 3]
```diff
@@ -273,13 +291,70 @@ void PoketchData_SetMapMarker(PoketchData *poketchData, int index, u8 x, u8 y);
...
+BOOL PoketchData_DotArtModified(const PoketchData *poketchData);
+
+/**
+ * Gets the current dot art data and copies it to the given dotArtData. This function will write 
+ * POKETCH_DOTART_SIZE_BYTES bytes at this pointer.
+ *
+ * @param poketchData:      The Poketch data to check.
+ * @param dotArtData:       The dot art data to copy into.
+ */
+void PoketchData_DotArtData(const PoketchData *poketchData, u8 *dotArtData);
```
Comment (cbt6) [inline include/poketch_data.h:311]:
```suggestion
void PoketchData_CopyDotArtData(const PoketchData *poketchData, u8 *dst);
```

File: include/poketch_data.h (hunk 1 above)
Comment (TrainerRiley) [reply include/poketch_data.h:52]: I wasn't really sure how to handle these constants, since the style guide says to prefer enums over #defines whenever possible.

**Comment by TrainerRiley** [comment]
Alright, switched those 4 enum values to defines and renamed the function parameters cbt mentioned.

**Comment by TrainerRiley** [comment]
Okay, I've removed a lot of redundant comments + improved some function names for clarity.

File: include/poketch_data.h [hunk 4]
```diff
@@ -49,245 +54,149 @@ enum PoketchScreenColor {
...
 
-/**
- * Gets the Pedometer's current step count value.
- *
- * @param poketchData:      The Poketch data to check.
- *
- * @return pedometer's step count.
- */
-u32  PoketchData_PedometerValue(const PoketchData *poketchData);
+u32 PoketchData_PedometerValue(const PoketchData *poketchData);
```
Comment (lhearachel) [inline include/poketch_data.h:134]: `PoketchData_StepCount`?

File: include/poketch_data.h [hunk 5]
```diff
@@ -49,245 +54,149 @@ enum PoketchScreenColor {
...
 
 /**
- * Sets the Poketch's pedometer value.
- *
- * @param poketchData:      The Poketch data to update.
- * @param value:            The step value to set.
+ * Overwrites the current step count. This function will not do anything unless the Pedometer app is registered.
  */
 void PoketchData_SetPedometerValue(PoketchData *poketchData, u32 value);
```
Comment (lhearachel) [inline include/poketch_data.h:139]: `PoketchData_SetStepCount`?

File: src/poketch_data.c [hunk 6]
```diff
@@ -236,73 +236,72 @@ void PoketchData_MapMarkerPos(const PoketchData *poketchData, int index, u8 *x,
...
-    poketchData->unk_00_2 = 1;
+    MI_CpuCopy8(src, poketchData->dotArtData, POKETCH_DOTART_SIZE_BYTES);
+    poketchData->dotArtModifiedByPlayer = TRUE;
 }
 
-void sub_02056A48 (PoketchData *poketchData, const BoxPokemon * param1)
+void PoketchData_PokemonHistoryAddEntry(PoketchData *poketchData, const BoxPokemon * boxPokemon)
 {
-    int v0;
+    int new_pokemon_index = PoketchData_PokemonHistorySize(poketchData);
```
Comment (lhearachel) [inline src/poketch_data.c:259]: I think you can just name this `idx` or `next`, and the meaning is perfectly clear.

File: src/poketch_data.c [hunk 7]
```diff
@@ -236,73 +236,72 @@ void PoketchData_MapMarkerPos(const PoketchData *poketchData, int index, u8 *x,
...
+void PoketchData_SetDotArtData(PoketchData *poketchData, const u8 *src)
 {
-    MI_CpuCopy8(param1, poketchData->unk_2A, 120);
-    poketchData->unk_00_2 = 1;
+    MI_CpuCopy8(src, poketchData->dotArtData, POKETCH_DOTART_SIZE_BYTES);
+    poketchData->dotArtModifiedByPlayer = TRUE;
 }
 
-void sub_02056A48 (PoketchData *poketchData, const BoxPokemon * param1)
+void PoketchData_PokemonHistoryAddEntry(PoketchData *poketchData, const BoxPokemon * boxPokemon)
```
Comment (lhearachel) [inline src/poketch_data.c:257]: Trailing space between `*` and `boxPokemon`

File: include/poketch_data.h [hunk 8]
```diff
@@ -49,245 +54,149 @@ enum PoketchScreenColor {
...
+    */
     struct {
-        u16 unk_00;
-        u16 unk_02;
-        u32 unk_04;
-    } unk_B8[12];
+        u16 species;
+        u16 icon;
+        u32 form;
+    } pokemonHistory[POKETCH_POKEMONHISTORY_MAX];   //!< The pokemon to display in the Pokemon History app
```
Comment (lhearachel) [inline include/poketch_data.h:99]: Maybe it makes sense to name this in a way that clarifies that the operations on it treat it like a queue with bounded size.

File: include/poketch_data.h [hunk 9]
```diff
@@ -49,245 +54,149 @@ enum PoketchScreenColor {
...
+BOOL PoketchData_DotArtModified(const PoketchData *poketchData);
+
+void PoketchData_DotArtData(const PoketchData *poketchData, u8 *dst);
+
+void PoketchData_SetDotArtData(PoketchData *poketchData, const u8 *src);
+
+/**
+ * Adds a new Pokemon to the end of the Pokemon History list.
+ */
+void PoketchData_PokemonHistoryAddEntry(PoketchData *poketchData, const BoxPokemon *boxPokemon);
```
Comment (lhearachel) [inline include/poketch_data.h:184]: `PoketchData_PokemonHistoryEnqueue`? That might not be clear to folks without a formal comp-sci background.

**Comment by TrainerRiley** [comment]
Addressed lhea's comments

**Review by lhearachel — APPROVED** [review]
Will merge pending OK build
